core/src/main/java/hudson/model/ItemGroupMixIn.java

```diff
@@ -271,7 +272,7 @@ public synchronized TopLevelItem createProjectFromXML(String name, InputStream x
         dir.mkdirs();
         boolean success = false;
         try {
-            XMLUtils.safeTransform((Source)new StreamSource(xml), new StreamResult(configXml));
+            XMLUtils.safeTransform((Source)new StreamSource(xml), new StreamResult(new FileOutputStream(configXml)));
```

Comment:
There is an underlying issue here.

> ` XMLUtils.safeTransform((Source)new StreamSource(xml), new StreamResult(configXml));`

This is not the correct way to construct a `StreamSource` from a String of XML content.

This should use the `StreamSource` constructor that takes either a `Reader` or an `InputStream`.